client/js/lounge.js

```diff
@@ -1045,8 +1041,12 @@ $(function() {
 	});
 
 	sidebar.on("click", "#sign-out", function() {
+		socket.emit("sign-out", storage.get("token"));
 		storage.remove("token");
-		location.reload();
+
+		if (!socket.connected) {
```

Comment:
Ah right, so if we are connected, we'll reload on another message, gotcha. I was wondering why we wouldn't reload if connected.